flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -142,6 +188,16 @@ protected abstract void submitRequestEntries(
      */
     protected abstract long getSizeInBytes(RequestEntryT requestEntry);
 
+    /**
+     * The {@code accept} method should be called on this Consumer if the processing of the {@code
+     * requestEntries} raises an exception that should not be retried. Specifically, any action that
+     * we are sure will result in the same exception no matter how many times we retry should raise
+     * a {@code RuntimeException} here. For example, wrong user credentials. However, it is possible
+     * intermittent failures will recover, e.g. flaky network connections, in which case, some other
+     * mechanism may be more appropriate.
+     */
+    protected final Consumer<Exception> fatalExceptionCons;
```

Comment:
Nit: I'd favor having a protected getter over the direct field access.

Reply:
I agree with this, since it would be a non-backwards compatible change to reduce scope once merged, let's fix this before merging

Reply:
Thank you! I've also moved this to the top of the class with the other fields.